src/main/java/com/stripe/model/Source.java

```diff
@@ -0,0 +1,216 @@
+		this.currency = currency;
+	}
+
+	public String getFlow() {
+		return flow;
+	}
+
+	public void setFlow(String flow) {
+		this.flow = flow;
+	}
+
+	public Boolean getLivemode() {
+		return livemode;
+	}
+
+	public void setLivemode(Boolean livemode) {
+		this.livemode = livemode;
+	}
+
+	public SourceOwner getOwner() {
+		return owner;
+	}
+
+	public void setOwner(SourceOwner owner) {
+		this.owner = owner;
+	}
+
+	public String getStatus() {
+		return status;
+	}
+
+	public void setStatus(String status) {
+		this.status = status;
+	}
+
+	public String getUsage() {
+		return usage;
+	}
+
+	public void setUsage(String usage) {
+		this.usage = usage;
+	}
+
+	public String getType() {
+		return type;
+	}
+
+	public void setType(String type) {
+		this.type = type;
+	}
+
+	// Flow-specific getters/setters
+
+	public SourceReceiverFlow getReceiver() {
+		return receiver;
+	}
+
+	public void setReceiver(SourceReceiverFlow receiver) {
+		this.receiver = receiver;
+	}
+
+	public SourceRedirectFlow getRedirect() {
+		return redirect;
+	}
+
+	public void setRedirect(SourceRedirectFlow redirect) {
+		this.redirect = redirect;
+	}
+
+	public SourceVerificationFlow getVerification() {
+		return verification;
+	}
+
+	public void setVerification(SourceVerificationFlow verification) {
+		this.verification = verification;
+	}
+
+	// Type-specific getters/setters
+
+	public Map<String, String> getTypeData() {
+		return typeData;
+	}
+
+	public void setTypeData(Map<String, String> typeData) {
+		this.typeData = typeData;
+	}
+
+	// APIResource methods
+
+	public String getSourceInstanceURL()
+		throws InvalidRequestException {
+		if (this.getCustomer() != null) {
+			return String.format("%s/%s/sources/%s", classURL(Customer.class), this.getCustomer(), this.getId());
+		} else {
+			return instanceURL(Source.class, this.getId());
+		}
+	}
+
+
+	public static Source create(Map<String, Object> params)
+			throws AuthenticationException, InvalidRequestException,
+			APIConnectionException, CardException, APIException {
+		return create(params, (RequestOptions) null);
+	}
+
+	public static Source create(Map<String, Object> params, RequestOptions options)
+			throws AuthenticationException, InvalidRequestException,
+			APIConnectionException, CardException, APIException {
+		return request(RequestMethod.POST, classURL(Source.class), params, Source.class, options);
+	}
+
+	public static Source retrieve(String id) throws AuthenticationException,
+			InvalidRequestException, APIConnectionException, CardException,
+			APIException {
+		return retrieve(id, (RequestOptions) null);
+	}
+
+	public static Source retrieve(String id, RequestOptions options)
+			throws AuthenticationException, InvalidRequestException,
+			APIConnectionException, CardException, APIException {
+		return request(RequestMethod.GET, instanceURL(Source.class, id), null, Source.class, options);
+	}
+
+	@Override
+	public Source verify(Map<String, Object> params) throws
```

Comment:
And sorry for the nitpickiness, but mind just moving all `throws` to appear on the next line for consistency?